lib/services/lwm2mHandlers/registration.js

```diff
@@ -190,12 +190,12 @@ function addUnsupportedAttributes(payload, configuration, deviceInformation, cal
             // In NGSIv2, spaces cannot be used in attributes' names. Therefore, we use percent-enconding.
             if (iotAgentLib.configModule.checkNgsi2()) {
```

Comment:
```suggestion
            // %23 is the URL encode char for '#', as that char is forbidden by the NGSIv2 spec in attribute names
            if (iotAgentLib.configModule.checkNgsi2()) {
```

I think that explanation helps to understand why this is doing here.